app/nginx-1.11.10/src/core/ngx_connection.c

```diff
@@ -13,6 +13,33 @@
 #if (NGX_HAVE_FSTACK)
 extern int fstack_territory(int domain, int type, int protocol);
 extern int is_fstack_fd(int sockfd);
+
+#define ngx_ff_skip_listening_socket(ls)                                      \
+    if (ngx_process <= NGX_PROCESS_MASTER) {                                  \
+                                                                              \
+        /* process master,  kernel network stack*/                            \
+        if (!ls[i].belong_to_host) {                                          \
+            /* We should continue to process the listening socket, */         \
+            /*    if it is not supported by fstack. */                        \
+            if (fstack_territory(ls[i].sockaddr->sa_family, ls[i].type, 0)) { \
+                continue;                                                     \
+            }                                                                 \
+        }                                                                     \
+    } else if (NGX_PROCESS_WORKER == ngx_process) {                           \
+        /* process worker, fstack */                                          \
+        if (ls[i].belong_to_host) {                                           \
+            continue;                                                         \
+        }                                                                     \
+                                                                              \
+        if (!fstack_territory(ls[i].sockaddr->sa_family, ls[i].type, 0)) {    \
+            continue;                                                         \
+        }                                                                     \
+    } else {                                                                  \
+        ngx_log_error(NGX_LOG_ALERT, cycle->log, 0,                           \
+                            "unexpected process type: %d, ignored",           \
+                            ngx_process);                                     \
+        exit(1);                                                              \
+    }                                                                   
 #endif
```

Comment:
Is it better to use a function than macro here?

Reply:
On this, function and macro are same, encapsulates the public operation to carry out code reuse. 
Macro maybe have a better performance compared with (inline) function.  
What do you think? Have I missed something?

Reply:
> On this, macro and function are same, encapsulates the public operation to carry out code reuse.
Macro maybe have a better performance compared with (inline) function.

You are right, they are same, and macro has a better performance, but it has effect, it's only called during initializing.
I just thought that the code is a little more here.